pkg/scannerv4/client/options.go

```diff
@@ -98,6 +98,9 @@ func SkipIndexerTLSVerification(o *options) {
 }
 
 // WithIndexerAddress specifies the gRPC address to connect.
+// This should be used when the client wants to reach out to both the Indexer and Matcher,
+// but they live at different addresses. When the client only wants to reach the Indexer or
+// both the Indexer and Matcher are at the same address, then use WithAddress.
```

Comment:
Nitpick: A better explanation of this option is:

```suggestion
// WithIndexerAddress specifies the gRPC address to connect to the Indexer service.
//
// Use this to direct the client to perform Indexer service calls to a different address from the default.
// Matcher service calls will be directed to the default address unless WithMatcherAdress is specified.  
```

Also, maybe we can add a comment to `WithAddress`:

```
// WithIndexerAddress specifies the gRPC address to connect.
//
// The address will be used for both Indexer service and Matcher service calls. To use different
// addresses for each service, use With{Indexer,Matcher}Address.
```